### ISSUE TITLE
Apply new click sounds to switches

### ISSUE THREAD

**Issue by gilbertohasnofb** [Juanvvc/c172p-detailed#238]
Apply new click sounds to cockpit switches (lights, master, avionics switch, etc.). Should be fairly soft.

Could you give me a hand with this, @onox?

**Comment by gilbertohasnofb** [comment]
These are the properties that, when changed, should activate this click sound:

controls/lightning/taxi-light
controls/lightning/landing-lights
controls/lightning/nav-light
controls/lightning/beacon
controls/lightning/strobe
controls/anti-ice/pitot-heat
controls/switches/master-avionics
controls/engines/engine/master-bat
controls/engines/engine/master-alt

**Comment by onox** [comment]
I'll do this in a separate branch, closing #85 without a PR felt so unnatural :confused: :smile:

**Comment by gilbertohasnofb** [comment]
No worries and no rush, do it the way you like better. I personally love to close the issues that I just solved myself :smile:

**Comment by onox** [comment]
hehe.. yeah, but new stuff should enter master via PR so it can be properly tested. But I think I have said this a thousand times before :grinning: :laughing:

**Comment by wkitty42** [comment]
:+1:

**Comment by gilbertohasnofb** [comment]
For when this gets implemented: we now have three click sound files in Sound, `click.wav`, `click2.wav` and `click3.wav`. My idea is:

- `click.wav` should be used for the small switches (taxi-light, landing-lights, nav-light, beacon, strobe and pitot-heat)
- `click2.wav` should be used for the master switch (master-bat and master-alt)
- `click3.wav` should be used by the avionics switch (master-avionics)

Since these files are super small, it makes no difference to have more variety, and it certainly brings some ear candy to the plane :wink:

**Comment by wkitty42** [comment]
haha! and here i thought the clicks were already implemented and i wasn't hearing them for some reason... was thinking something was wrong over here but had not yet written up an issue... i'll wait for them to be implemented :wink:

**Comment by onox** [comment]
I'll try to apply the new sounds tonight or tomorrow.

**Comment by wkitty42** [comment]
:+1: no problem! :laughing:

**Comment by gilbertohasnofb** [comment]
@onox ok, let me know how it goes and if you need anything that I can help with related to the sounds.

**Comment by onox** [comment]
You could help me by finding an ambient/background sound of "something floating on the water", but that's not related to this bug :smile:

**Comment by gilbertohasnofb** [comment]
> You could help me by finding an ambient/background sound of "something floating on the water"

Sure! You mean a sound of something moving through water or simply floating still? (which begs the question, do things floating still make any sound?). For this, let's use https://github.com/Juanvvc/c172p-detailed/issues/264

**Comment by onox** [comment]
@wlbragg Are you able to determine (x,y,z) coordinates of objects in the 3D model?

**Comment by onox** [comment]
@gilbertohasnofb Which sound do you want for the circuit breakers?

**Comment by onox** [comment]
More dials and switches that could use sounds:

1. Primer, throttle, mixture, air heat, air flow
2. Flaps
3. Magneto's
4. Circuit breakers
5. Parking brake
6. Carb heat
7. Dials: instrument lighting, attitude horizon offset, heading offset, heading bug, altimeter, nav1/nav2/adf radial/heading
8. Doors

**Comment by gilbertohasnofb** [comment]
> Which sound do you want for the circuit breakers?

Please use `click.wav`

> More dials and switches that could use sounds

I will work on them all and report here.

**Comment by onox** [comment]
@gilbertohasnofb I think the circuit breakers should have a sound that sounds like a button being pushed in and then moves little bit back.

**Comment by gilbertohasnofb** [comment]
Okay, working on it then.

**Comment by onox** [comment]
Can you add any new sounds for this bug to the bug-238 branch? Do `git fetch` and then `git checkout bug-238`.

**Comment by gilbertohasnofb** [comment]
Sure onox, I can do that, but also let me ask you something: I completely understand why you ask people dealing with code to use PRs or different branches to upload their work, but does this make any sense when comes to textures and sounds? I mean, they can't break anything and testing means simply looking or hearing, right? Or am I missing something here? I am asking this very sincerely as you know much more about git than I do and very likely I am missing something..

**Comment by onox** [comment]
Well, otherwise I have to merge master to bug-238 all the time to get the sounds when you make a change to the sounds and that will make the graph (history) more complicated than necessary.

**Comment by gilbertohasnofb** [comment]
Oh, I see. But in the general case of textures, I can continue to simply push them to the master, right?

**Comment by onox** [comment]
But the textures and sounds don't break anything, that's correct. About `water-fast.wav` and `water-still.wav`: that's ok to add to master directly because I haven't started working on #264 yet.

**Comment by onox** [comment]
Yes.

**Comment by gilbertohasnofb** [comment]
Cool, thanks! I was meaning to ask you this for some long time but it had slipped my mind :smile: So the next sounds for the switches will go to `bug-238`

**Comment by onox** [comment]
Yeah, just the sounds for the switches :smile: Unrelated sounds can go to master.

**Comment by onox** [comment]
@gilbertohasnofb Have you already committed any sounds locally? Otherwise I can commit the circuit breakers in the mean time.

**Comment by gilbertohasnofb** [comment]
Everything is ready, I am uploading them now

**Comment by gilbertohasnofb** [comment]
onox, this `git fetch` is taking ages. Would it be all right if I simply push these files to the master? Sorry but my internet is not cooperating today and I will have to wait the whole night for it to finish fetching if it keeps at this speed...

**Comment by onox** [comment]
Can you wait for fetch 5 more minutes? :open_mouth:

**Comment by gilbertohasnofb** [comment]
Ok, let's try a bit longer, but right now it's on 23% and I am waiting for almost 10 minutes, and soon I need to go (I am a night owl but tomorrow I must wake up early). How large is the whole repository, would you know?

**Comment by onox** [comment]
I don't know, but you have a recent version, right?

**Comment by gilbertohasnofb** [comment]
Well, normally I am just pulling the master and it was up to date, but I don't ever use git fetch. Since I never use different branches I normally just pull, then add, commit and push.

**Comment by onox** [comment]
Ok, can you cancel `git fetch` and then just try to do `git checkout bug-238`?

**Comment by gilbertohasnofb** [comment]
Just did that, I get: `error: pathspec 'bug-238' did not match any file(s) known to git.`

Alternatively, I could also upload it on some sharing service so you can download and push it yourself. Sorry onox, but my internet is really bad at home. Tomorrow at my university I will fetch the whole repository again.

**Comment by onox** [comment]
What if you do `git pull`?

**Comment by gilbertohasnofb** [comment]
Hmm, before it was telling me I was up to date, but since I cancelled the `git fetch` I now need to download 5000 files.

**Comment by onox** [comment]
Uh? Just do `git fetch` tomorrow then on the university. I need to sleep too :smiley: :zzz:

**Comment by onox** [comment]
And then just add those sounds to bug-238. Should be no problem if `git fetch` succeeds.

**Comment by gilbertohasnofb** [comment]
Ok onox, so I will push those sounds tomorrow in the afternoon then. Thanks a lot!

**Comment by onox** [comment]
Oh wait.

**Comment by gilbertohasnofb** [comment]
yes

**Comment by onox** [comment]
Can you do `git checkout origin/bug-238`?

**Comment by gilbertohasnofb** [comment]
`error: pathspec 'origin/bug-238' did not match any file(s) known to git.` :cry:

**Comment by gilbertohasnofb** [comment]
Tomorrow I will clone the whole repo again from at uni to make sure everything is working well, then I push those sounds.

**Comment by onox** [comment]
Ok, nvm, just do `git fetch` tomorrow when you have fast internet. No worries :+1: :smile:

**Comment by gilbertohasnofb** [comment]
Ok, to be continued then!

**Comment by wkitty42** [comment]
wow! 5000 files?? something isn't right... i don't know what but i've not had to pull 5000 files for anything...

**Comment by gilbertohasnofb** [comment]
> wow! 5000 files?? something isn't right...

Yes it is. Try cloning the repository into a new folder and you'll see that the total number of files is around 5000. Normally a pull just grabs what is new, but something went wrong with my directory and I am cloning it again.

**Comment by gilbertohasnofb** [comment]
@onox the new files are pushed. Here is what to do with them:

1. Primer, throttle, mixture, air heat, air flow: no sound (I don't think they make any noticiable sound, at least I couldn't hear it watching videos of the aircraft"
2. Flaps: `keys2.wav`
3. Magnetos: `keys.wav` and `keys2.wav` (can we randomize them or alternate or at least assign one to L and BOTH and the other to OFF and R magnetos?)
4. Circuit breakers: `double-click.wav` (should be really really soft)
5. Parking brake: `lever.wav`
6. Carb heat: no sound
7. Dials (instrument lighting, attitude horizon offset, heading offset, heading bug, altimeter, nav1/nav2/adf radial/heading): `scroll.wav`
8. Doors: `door-opening.wav` and `door-closing.wav` (remember to position it left and right according to which door is being used)
9. Trim wheel: `scroll.wav`
10. Buttons and dials of radio stack:  use `double-click2.wav` for buttons and `scroll.wav` for dials

And one of the sounds is CC attribution, so we need to add this to some sort of "thanks" file (what's the best way of doing that?):
`- DrMinky for the "Old Lever Pull" sound: http://freesound.org/people/DrMinky/sounds/167048/`

**Comment by onox** [comment]
@gilbertohasnofb I am going to work on the sounds now :smiley:

**Comment by onox** [comment]
Parking brake, magneto's+starter, and flaps do not make any sound when using the keyboard yet.

**Comment by gilbertohasnofb** [comment]
Hi onox, I just tested your branch and here are some comments:

- I can't hear flap sounds even when clicking on them
- the dial sounds are too loud IMO. I'd make them barely audible when the engine is at high RPM.
- the masters switch sound (both ALT and BAT) are a bit buggy. Try clicking on them constantly (let's say 3 times a second): sometimes they will play fine but sometimes they will play a broken sound (quite unpleasant) 
- light switches also got a bit buggy: in the beginning they were fine, but after I started the engine, they were sometimes making proper sounds and sometimes silent! They were almost alternating between sound/no-sound at one point

**Comment by gilbertohasnofb** [comment]
I also added a phrase to the `Thanks` file about the parking break sound.

**Comment by wkitty42** [comment]
that's for the parking brake, right? :wink:

**Comment by onox** [comment]
> I can't hear flap sounds even when clicking on them

I can definitely hear a click sound when I click on the flaps lever, but you'll also hear the sound of the flaps itself. The sound of the flaps is positioned near the speaker in the top panel. Is that where it should come from? Or should it come from the left and right wings?

> the dial sounds are too loud IMO. I'd make them barely audible when the engine is at high RPM.

I'll reduce the volume.

> the masters switch sound (both ALT and BAT) are a bit buggy.
> light switches also got a bit buggy

I have no problems with the switches. If you toggle `/sim/model/c172p/sound/click-master` to true, the sound will play, and when you toggle it to false and then to true again, you can re-play the sound again. The Nasal code will toggle the property to false after 0.1 seconds, which is the more than the duration of these click sounds. You would have to click faster than 10 times per second if you want to interrupt the sounds. You can try to toggle `/sim/model/c172p/sound/click-parking`, which will play a second that takes 1.6 seconds. If you toggle it to false the sound will simply stop, if you toggle it to true again, the sound will play from the beginning again. I can't hear any brokenness.

**Comment by onox** [comment]
@gilbertohasnofb I've reduced the volume of the dial sounds.

**Comment by gilbertohasnofb** [comment]
> I can definitely hear a click sound when I click on the flaps lever

You are correct, the sound is definitely there. As for the position, I really don't know.

>  I've reduced the volume of the dial sounds.

Much better IMO!

> You would have to click faster than 10 times per second if you want to interrupt the sounds.

I hear some weird effect sometimes at a much slower rate than this. I tried to record it but the video is horrible. Anyway, see if you can hear it (turn up your volume as the video is very soft). At around 00:07 you can hear the first times the sound breaks, and at around 00:14 again. Link: https://www.youtube.com/watch?v=mm3bInvy_hc&feature=youtu.be

**Comment by onox** [comment]
I can hear the clicks in the video, but I don't hear anything wrong with them :confused:

**Comment by gilbertohasnofb** [comment]
@onox Yeah, I was afraid of that. Unfortunately the program I use to record video is rubbish when it comes to recording sound. Let's just continue implementing this and in the meantime I will make some tests on my side and see if I find anything reliable on how to reproduce these things I mentioned, ok?

@wkitty42 
> that's for the parking brake, right?

Yep! Sorry for my EngRish :smile: I already corrected the `Thanks` file.

@tigert Tuomas, which kind of sound does a trim wheel make? Would you be able to describe it?

**Comment by onox** [comment]
@gilbertohasnofb I see you added a commit to the bug-238 branch you also added to the master branch? If you add it to the master branch, then there's no need to add the same commit to this feature branch because we only care about adding sounds in this branch.

**Comment by gilbertohasnofb** [comment]
@onox Yeah, it was my mistake and I realised as soon as I did it (you can note that I added it first to the bug-238 branch; I had forgotten which branch I was in, sorry). But since it was a harmless addition to that branch, I just left things as they were. Is that a problem? Sorry for the mess..

**Comment by onox** [comment]
No it's not a problem.

**Comment by onox** [comment]
@gilbertohasnofb I have added the sounds for the radio stack, except the two comm/nav radio's and the transponder. Some issues with the radio stack:

1. The two nav radio's and transponder live in: `Aircraft/Instruments-3d/kt76a/kt76a.xml`, `Aircraft/Instruments-3d/kx165/kx165-1.xml`, and `Aircraft/Instruments-3d/kx165/kx165-2.xml`, so I couldn't add sounds for them.

2. The buttons of the autopilot (kap140) instrument do not move :cry: 

3. We are currently using just the first 0.1 seconds of `double-click2.wav`. The full sound is 0.25 seconds long. Using the full part doesn't make sense because you can keep a (mouse)button pressed.

    Maybe the ADF, radio's, and AP instruments can be modified, so that buttons move from their 0.0 position to 1.2 and then back to 1.0 if you want to enable the button. (If that's how the buttons work) And move from 1.0 to 1.2 and then back to 0.0 if you want to disable it. In that case it might make sense to split `double-click2.wav` into two parts.

Notes:
Position comm1/nav1:

-0.3660
-0.12
-0.04

Position comm2/nav2:

-0.3660
-0.12
-0.09

Position transponder:

-0.3660
-0.12
-0.25

**Comment by onox** [comment]
@gilbertohasnofb I also think that a different sound than `scroll.wav` should be used for the trim wheel. The autopilot might move the trim wheel and then it looks like some dial is being moved. It sounds confusing :stuck_out_tongue_closed_eyes:

**Comment by onox** [comment]
I think if you can add a new sound for the trim wheel scroll, then we can merge PR #265. The issues above require copy and modifying files from fgdata.

**Comment by onox** [comment]
@gilbertohasnofb I think your commit actually caused the branch to not be automatically be merge-able. However, I have done some interactive rebasing to squash some commits and simply removed your particular commit in the process.

**Comment by gilbertohasnofb** [comment]
@onox I will try to add a new sound for the trim wheel by tonight.

About the nav radios and transponder, why don't we copy them into our project then? Together that's only 439 kb more.

As for my last erroneous commit on this branch, my apologies again. Glad to hear it's mergeable right now.

**Comment by gilbertohasnofb** [comment]
@onox Okay, so I have prepared the new sound for the trim tab, but I am having a problem with our branch. I still didn't even try to add nor commit nor push this time. I simply did a git pull on the master branch (all fine), then when I did a `git checkout bug-238` I got:

		Switched to branch 'bug-238'
		Your branch and 'origin/bug-238' have diverged,
		and have 17 and 13 different commits each, respectively.
			(use "git pull" to merge the remote branch into yours)

So I tried `git pull` again, but this time I got:

		Auto-merging c172p-set.xml
		CONFLICT (content): Merge conflict in c172p-set.xml
		Auto-merging c172-sound.xml
		CONFLICT (content): Merge conflict in c172-sound.xml
		Automatic merge failed; fix conflicts and then commit the result.

And if I another `git pull`, I get:

		M	Models/Interior/Panel/Instruments/AI/AI.xml
		M	Models/Interior/Panel/Instruments/dme/dme.xml
		M	Models/Interior/Panel/Instruments/dme/ki266.xml
		M	Models/Interior/Panel/Instruments/kap140/KAP140TwoAxisAlt.xml
		M	Models/Interior/Panel/Instruments/kma20/kma20.xml
		M	Models/Interior/Panel/Instruments/kr87-adf/kr87.xml
		U	c172-sound.xml
		U	c172p-set.xml
		Pull is not possible because you have unmerged files.
		Please, fix them up in the work tree, and then use 'git add/rm <file>'
		as appropriate to mark resolution, or use 'git commit -a'.

Do you know what is happening? If you can give me a hand solving this then I can push the new sound file as soon as possible.

**Comment by wlbragg** [comment]
@gilbertohasnofb, onox is definitely the one to sort this out but I can tell you on the second pull you had conflicts that required resolution (meaning you need to edit those files and correct the "tagged" lines that are in conflict). Then you commit those conflict resolution changes you just fixed.and you are ready to merge again. 
The third pull is simply telling you that you have files that aren't merged (all your new ones marked with "M") and the 2 files you need to edit that had conflicts (marked U).

**Comment by gilbertohasnofb** [comment]
Thanks @wlbragg. So if I manually copy the raw data from those two xml files from the but-238 branch here on GitHub and paste it on my repository, then I would have solved the conflict?

**Comment by wlbragg** [comment]
The "two files" being
U   c172-sound.xml
U   c172p-set.xml
?

**Comment by gilbertohasnofb** [comment]
Yep.

**Comment by wlbragg** [comment]
Yes, that sounds right.

**Comment by wlbragg** [comment]
Is your local branch older than #238 PR?

**Comment by wlbragg** [comment]
That is why when you get a conflict notice you edit the files that are flagged in conflict. Because technically you should be able to tell what part to keep and what part to discard in the conflict.
Have you ever seen or edited one of those conflicts?

**Comment by gilbertohasnofb** [comment]
> Is your local branch older than #238 PR?

No, I re-cloned the whole repository some days ago.

> Have you ever seen or edited one of those conflicts?

Not really.

> Yes, that sounds right.

I will give it a shot now then.

**Comment by wlbragg** [comment]
"should be able to tell what part to keep and what part to discard in the conflict"
Because you might be ahead of the PR or behind the PR and that makes a difference in what conflict part is discarded.
If you are ahead , you may want to keep your version of the conflict, if behind, you may want to keep what is in 238.

**Comment by wlbragg** [comment]
Sometime if I really can't or don't want to wait for help, I'll reclone a fresh copy of master, make my changes and push that back to github (that is if the changes aren't too extensive..
I have the cpu power speed and bandwidth to allow me to do that in a reasonable amount of time.

**Comment by wlbragg** [comment]
But then again I guess you trying to work with a PR so that is a bit different.

**Comment by gilbertohasnofb** [comment]
I really don't know then, and copying the raw data didn't solve it, the conflict is still present... 

> Because you might be ahead of the PR

How could I be ahead of the PR? I myself never touch those xml files, I only add wav files. 

> Sometime if I really can't or don't want to wait for help, I'll reclone a fresh copy of master

I do the same when I am at my uni, but at home my internet is rubbish. This repository is quite large nowadays, so I am hoping to solve it somehow else.

**Comment by wlbragg** [comment]
Open the files that have the conflict and find the conflict, take a look at it and see if you recognize both versions of the conflict.

**Comment by wlbragg** [comment]
There may be more than one conflict in each file.

**Comment by gilbertohasnofb** [comment]
Sure, I already checked with a difference checker, it's just some few lines of code per file. But the point is that simply correcting these lines on the files is not solving the conflict...

You know what, I need to do some other stuff that does not involve internet right now, so I will just reclone the whole repository. This is too much headache for me. Thanks anyway for the help!

**Comment by wlbragg** [comment]
Good luck!

**Comment by gilbertohasnofb** [comment]
Thanks, with my bandwidth I will need it! :wink:

**Comment by gilbertohasnofb** [comment]
@wlbragg okay, it took me 2h but cloning is finished and problem is solved. Thanks again!

@onox so the new trim sound is pushed, named `trim.wav`. It should be really soft, and the sound is loopable. Let me know if we still need some work on these sounds, okay?

**Comment by onox** [comment]
I did an interactive rebase as I said in https://github.com/Juanvvc/c172p-detailed/issues/238#issuecomment-111710053 and force push to clean up this feature branch a bit, but you might not have understood the consequences of it . So what happens is that your commit tree is something like:

`A->B->C->D->E->F->G->H`

while mine (and Github) is:

`A->B->I->J->K`

What you should do in this case is find the most common ancestor, which is in this case commit B. Then you do `git reset --hard B` and then you do `git pull` to get commits I, J, and K. That will save you having to do a full clone.

Perhaps I should have added these instructions on the PR page, but I didn't.

**Comment by onox** [comment]
@gilbertohasnofb I get:

```
AL Error (sound manager): Invalid Enum at buffer add data
No such buffer!
```

Could it be because the sound is 24 bit instead of 16 bit? :confused:

**Comment by onox** [comment]
I've modified `c172-sound.xml` to use `trim.wav`, so you can check if the sound is accepted by FG or not.

**Comment by gilbertohasnofb** [comment]
> I did an interactive rebase as I said in #238 (comment) and force push to clean up this feature branch a bit, but you might not have understood the consequences of it 

Yeah, I didn't realise it at all :smile but no worries, now everything is okay.

As for your error, I will convert the sound to 16-bit tomorrow and also test your last work.

**Comment by onox** [comment]
> About the nav radios and transponder, why don't we copy them into our project then? Together that's only 439 kb more.

I'll copy the files.

**Comment by onox** [comment]
@gilbertohasnofb Done.

**Comment by gilbertohasnofb** [comment]
Great!

**Comment by gilbertohasnofb** [comment]
Great news, I contacted the authors of the two CC-BY sound files we were using and they authorized the FlightGear project to include them as CC0 (public domain). I updated the Thanks file to reflect this (so now this Thanks file is not a license requirement but only a polite way of thanking who deserves credit and no license conflict will ever be inflicted). Now I have to solve only one issue with a single texture I used as source and then our project will be completely GPL or public domain (meaning that even if that licensing discussion going on in the mailing list decides to completely block CC-BY, we are clear to go).
